core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1084,7 +1084,7 @@ class ReplicaManager(val config: KafkaConfig,
       fetchInfos.foreach { case (topicPartition, partitionData) =>
         logReadResultMap.get(topicPartition).foreach(logReadResult => {
           val logOffsetMetadata = logReadResult.info.fetchOffsetMetadata
-          fetchPartitionStatus += (topicPartition -> FetchPartitionStatus(logOffsetMetadata, partitionData))
+          fetchPartitionStatus += (topicPartition -> FetchPartitionStatus(logOffsetMetadata, partitionData, logReadResult.divergingEpoch.nonEmpty))
```

Comment:
Hmm.. If the `LogReadResult` has a diverging epoch, wouldn't we want to respond immediately?

Reply:
ah, yes, so we don't need to check the original result in DelayedFetch, we return immediately here. Updated.